draft-jmiller-json-web-proof.md

```diff
@@ -67,29 +67,47 @@ There are a growing number of these cryptographic primitives that support select
+
+A JWP is initially created by the issuer using a `sign` interaction with an implementation.  A successful results is a signed JWP that has a single issuer-protected header, one or more payloads, and an initial proof value that contains the signing algorithm output.  The holder upon receiving a signed JWP then uses `validate` to check the integrity protection of the header and all payloads using the given proof value.
 
-The intent of JSON Web Proofs is to establish a common container format for multiple payloads that can be integrity-verified against a cryptographic proof value also in the container.  It does not create or specify any cryptographic protocols, interaction protocols, or custom options for algorithms with additional capabilities.
+After validation, the holder uses `present` to apply any selective disclosure choices, perform privacy-preserving transformations for unlinkability, and add a presentation-protected header that ensures the resulting presented JWP cannot be replayed.  The verifier then uses `verify` to ensure the integrity protection of the protected headers and any disclosed payloads along with verifying any additional ZKPs covering non-disclosed payloads.
+
+While `sign` and `validate` only occur when a JWP is initially created by the issuer, the `present` and `verify` steps may be safely repeated by a holder on a signed JWP.  The unlinkability of the resulting presented JWP is only protected when supported by the underlying algorithm.
 
 Algorithm definitions that support JWPs are being done in separate companion specifications - just as the [JSON Web Algorithms] [@RFC7518] specification does for JWS and JWE [@RFC7516].  The JSON Proof Algorithms specification defines how an initial set of algorithms are used with JWP.
 
 # JWP Format
 
-A JWP always contains a protected header along with one or more payloads.  Each payload is represents an octet string.  The payloads are then serialized as an ordered array.  Finally, the JWP contains a representation of a cryptographic proof over the payloads.
+A JWP may have one or two integrity protected headers depending on if it is the signed form (one) or the presented form (two).  The protected headers are always a JSON object that is serialized as a UTF-8 encoded octet string.
+
+Every JWP has one or more payloads, each payload is an octet string.  All of the contained payloads are serialized as an array who's ordering is preserved in all serializations.
+
+The JWP proof value is a single octet string that is only generated from and consumed by the underling JPA.  Internally the proof value may one or more cryptographic statements that are used to check the integrity protection of the issuer header and all payloads.  Each of these statements may be a ZKP or a traditional cryptographic signature, the algorithm is responsible for how these statements are serialized into a single proof value.
```

Comment:
Add a comma after "Internally".
Break the last sentence into two sentences at the comma.